Add recaptcha

denys-medynskyi opened the pull request:
Resolves #541

### Description

Adds Google recaptcha on user sign up form.
To manual test it you need tokens. I have them if you need.
No specs 😿

denys-medynskyi committed:
Add recaptcha on sign up form

sveredyuk commented on app/controllers/users/registrations_controller.rb:
```diff
@@ -32,5 +35,23 @@ def sign_up_params
+    end
+
+    def recaptcha_error_message
+      @recaptcha_error_message
```
could be simple attr_reader

sveredyuk commented on app/controllers/users/registrations_controller.rb:
```diff
@@ -1,5 +1,8 @@
 module Users
   class RegistrationsController < Devise::RegistrationsController
+    prepend_before_action :check_captcha, only: [:create]
```
`only: :create`

floor114 commented on app/controllers/users/registrations_controller.rb:
```diff
@@ -32,5 +35,23 @@ def sign_up_params
     end
+
+    def check_captcha
+      unless verify_recaptcha
```
`return if verify_recaptcha`

denys-medynskyi committed:
Add recaptcha to the reset password form

denys-medynskyi commented:
@sveredyuk @floor114 @kushniryb please take a look again. 
Also added recaptcha to the forgot password form, so I extracted shared thing to the module.
